braindecode/models/eegitnet.py

```diff
@@ -0,0 +1,313 @@
+            groups=in_channels,
+            bias=bias,
+            padding_mode=padding_mode,
+        )
+
+
+class EEG_ITNET(nn.Module):
+    """ EEG-ITNet: An Explainable Inception Temporal
+     Convolutional Network for motor imagery classification from
+     Salami et. al 2022.
+
+    See [Salami2022]_ for details.
+
+    Code adapted from https://github.com/abbassalami/eeg-itnet
+
+    Parameters
+    ----------
+    n_classes: int
+        number of outputs of the decoding task (for example number of classes in
+        classification)
+    n_in_chans: int
+        number of input EEG channels
+    input_window_samples : int
+        Number of time samples.
+    drop_prob: float
+        Dropout probability.
+
+    References
+    ----------
+    .. [Salami2022] A. Salami, J. Andreu-Perez and H. Gillmeister, "EEG-ITNet: An Explainable
+    Inception Temporal Convolutional Network for motor imagery classification," in IEEE Access,
+    doi: 10.1109/ACCESS.2022.3161489.
+    """
+
+    def __init__(self, n_classes, in_channels, input_window_samples,
+                 drop_prob=0.4):
+        super().__init__()
+        self.block11 = nn.Sequential(
+            *[
+                nn.Conv2d(
+                    1, 2, kernel_size=(1, 16), padding="same", bias=False
+                ),
+                nn.BatchNorm2d(2),
+                DepthwiseConv2d(
+                    2,
+                    kernel_size=(in_channels, 1),
+                    depth_multiplier=1,
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(2),
+                nn.ELU(),
+            ]
+        )
+        self.block12 = nn.Sequential(
+            *[
+                torch.nn.Conv2d(
+                    1, 4, kernel_size=(1, 32), padding="same", bias=False
+                ),
+                nn.BatchNorm2d(4),
+                DepthwiseConv2d(
+                    4,
+                    kernel_size=(in_channels, 1),
+                    depth_multiplier=1,
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(4),
+                nn.ELU(),
+            ]
+        )
+        self.block13 = nn.Sequential(
+            *[
+                torch.nn.Conv2d(
+                    1, 8, kernel_size=(1, 64), padding="same", bias=False
+                ),
+                nn.BatchNorm2d(8),
+                DepthwiseConv2d(
+                    8,
+                    kernel_size=(in_channels, 1),
+                    depth_multiplier=1,
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(8),
+                nn.ELU(),
+            ]
+        )
+
+        self.pool1 = nn.Sequential(
+            *[nn.AvgPool2d(kernel_size=(1, 4)), nn.Dropout(drop_prob)]
+        )
+
+        # ================================
+
+        self.block2 = nn.Sequential(
```

Comment:
`self.block2` to `self.block9` look all the same, my idea to simplify that: These pairs of blocks (e.g., `block2` and `block3`) create a TC block (terminology from the paper). This has a residual connection. There are several TC blocks that differs only due to `dilation`. We could create a class `_TCBlock` that creates two `_DepthWiseConv2D` blocks and implement residual connection in the forward method.